python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -25,39 +28,70 @@ def __init__(self, context):  # pylint: disable=too-many-locals
         # Extract parameters from resource config. Note that we can't pass None values to
         # snowflake.connector.connect() because they will override the default values set within the
         # connector; remove them from the conn_args dict.
-        self.conn_args = {
-            k: context.resource_config.get(k)
-            for k in (
-                "account",
-                "user",
-                "password",
-                "database",
-                "schema",
-                "role",
-                "warehouse",
-                "autocommit",
-                "client_prefetch_threads",
-                "client_session_keep_alive",
-                "login_timeout",
-                "network_timeout",
-                "ocsp_response_cache_filename",
-                "validate_default_parameters",
-                "paramstyle",
-                "timezone",
-            )
-            if context.resource_config.get(k) is not None
-        }
+        self.connector = context.resource_config.get("connector", None)
+
+        if self.connector == 'sqlalchemy':
+            self.conn_args = {
+                k: context.resource_config.get(k)
+                for k in (
+                    "account",
+                    "user",
+                    "password",
+                    "database",
+                    "schema",
+                    "role",
+                    "warehouse",
+                    "cache_column_metadata",
+                    "numpy",
+                )
+                if context.resource_config.get(k) is not None
+            }
+
+        else:
+            self.conn_args = {
+                k: context.resource_config.get(k)
+                for k in (
+                    "account",
+                    "user",
+                    "password",
+                    "database",
+                    "schema",
+                    "role",
+                    "warehouse",
+                    "autocommit",
+                    "client_prefetch_threads",
+                    "client_session_keep_alive",
+                    "login_timeout",
+                    "network_timeout",
+                    "ocsp_response_cache_filename",
```

Comment:
Are these arguments not applicable to snowflake connections made through sqlalchemy?

Reply:
I don't believe these other parameters are applicable. Here is the documentation that lists the connection parameters for Snowflake SQLAlchemy: https://docs.snowflake.com/en/user-guide/sqlalchemy.html#connection-parameters